### PR TITLE
DOC: Fixed PeriodArray api ref

### PR DESCRIPTION
**Pull request by TomAugspurger**
[ci skip]

DatetimeArray was listed twice. PeriodArray was missed.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
The linter complained about the ref to `pandas.PeriodIndex`. In the See Also.

I'm not sure if just `PeriodIndex` will give the correct reference to `pandas.PeriodIndex`. It's been a while since I looked at the discovery rules, but I thought that would only work when you're in the same `.. currentmodule`.  I believe that this is inside a `pandas.arrays` currentmodule, so this may fail.

File: pandas/core/arrays/period.py [hunk 1]
```diff
@@ -104,13 +104,15 @@ class PeriodArray(dtl.DatetimeLikeArrayMixin, dtl.DatelikeOps):
         is an ndarray of integers, when `freq` is required. When `values`
         is a PeriodArray (or box around), it's checked that ``values.freq``
         matches `freq`.
+    dtype : PeriodDtype, optional
+        A PeriodDtype instance from which to extract a `freq`.
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/period.py]: Does it override the `freq` keyword?

**Comment by TomAugspurger** [comment]
We raise when they're both specified but differ.

**Comment by jorisvandenbossche** [comment]
> We raise when they're both specified but differ.

Can you then mention this in the docstring you added?

File: pandas/core/arrays/period.py [hunk 2]
```diff
@@ -104,13 +104,16 @@ class PeriodArray(dtl.DatetimeLikeArrayMixin, dtl.DatelikeOps):
         is an ndarray of integers, when `freq` is required. When `values`
         is a PeriodArray (or box around), it's checked that ``values.freq``
         matches `freq`.
+    dtype : PeriodDtype, optional
+        A PeriodDtype instance from which to extract a `freq`. If both
+        `freq` and `dtype` are specified, then the frequencies must match.
     copy : bool, default False
```
Comment (jreback) [inline pandas/core/arrays/period.py]: so maybe a Raises section showing IncompatibleFrequecy

**Comment by jreback** [comment]
thanks @TomAugspurger

**Pull request merged**